mne_realtime/lsl_client.py

```diff
@@ -81,10 +86,13 @@ def get_data_as_epoch(self, n_samples=1024, picks=None):
     def iter_raw_buffers(self):
         """Return an iterator over raw buffers."""
         while True:
-            samples, _ = self.client.pull_chunk(max_samples=self.buffer_size)
-            if not len(samples):
-                samples = np.empty((self.info['nchan'], 0))
-            yield np.vstack(samples).T
+            _, timestamps = self.client.pull_chunk(
+                max_samples=self.buffer.shape[0],
+                dest_obj=self.buffer,
+            )
+            num_timestamps = len(timestamps) if timestamps else 0
+            data = self.buffer[:len(timestamps), :]
+            yield data.copy()
```

Comment:
i think this is the line @jasmainak was mentioning that could simply be `yield data` instead

Reply:
Thanks! Yeah, I hadn't been sure whether to copy here.

I saw that https://mne.tools/stable/overview/design_philosophy.html#in-place-operation prefers not-copying, and at the same time, I noticed that the previous implementation copied (via `np.vstack`, so I wasn't sure if not-copying could cause confusing errors for existing users.

Any preferences? I am new to contributing to MNE.

Reply:
We only need to `copy` here if we think *the user* later doing some inplace operation is problematic. In this case I think it could be. Let's say they take two chunks of data via iteration, then want to do something with them. If we *don't* `.copy` here, the `self.buffer` is the same `ndarray` so those two buffers will be identical, and equal to the second chunk of data. So I think the code here is right/safer than returning un-copied data.

Reply:
yeah, I think the not copying is good here. some of the code here was written before the design philosophy was applied.

Reply:
good point. I think with some of the online filtering and transformations might require that we copy here to be sure

Reply:
The reason I suggested removing the `copy` was that it might be slow which might be important in real-time setting. So yes, a tradeoff between doing the safe thing for the user vs optimizing for real-time ... which seemed to be the purpose of this PR?